dev_config/python/.pre-commit-config.yaml

```diff
@@ -7,19 +7,6 @@ repos:
       - id: check-yaml
       - id: debug-statements
 
-  - repo: https://github.com/PyCQA/flake8
-    rev: 7.0.0
-    hooks:
-      - id: flake8
-        args: ['--select=Q000']  # Q000 is the error code for single quote enforcement
-        additional_dependencies:
-          - flake8-quotes
-
-  - repo: https://github.com/hhatto/autopep8
```

Comment:
If Ruff, with @dorbanianas' suggestion applied, could also enforce single quote behavior, it will be great to remove flake8 check - we introduced it only for this reason.

Autopep8 is something different. Since Ruff doesn't fully cover its checks, I feel like we shall not remove it without compelling reasons.

Reply:
adding it as https://github.com/astral-sh/ruff/discussions/9743